common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -713,6 +718,10 @@ def _internal_depth_first(item_location, is_root):
             bulk_record = self._get_bulk_ops_record(location.course_key)
             bulk_record.dirty = True
             self.collection.remove({'_id': {'$in': to_be_deleted}})
+
+        # Now it's been published, add the object to the courseware search index so that it appears in search results
+        self.do_index(location)
```

Comment:
Great point @chrisndodge. It seems much cleaner to me to be firing model changed notifications from the modulestore, and then search can just be bolted on rather than being wired in directly.